src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -882,5 +882,53 @@ namespace cryptonote
       return ::serialization::parse_binary(graft_extra.data, graft_tx_extra);
   }
 
+  namespace
+  {
+      struct GraftStakeTxExtra
+      {
+          std::string SupernodePublicId;
+          cryptonote::account_public_address SupernodePublicAddress;
+
+          GraftStakeTxExtra() = default;
+          GraftStakeTxExtra(const std::string &supernode_public_id,
+            const cryptonote::account_public_address& supernode_public_address) :
+              SupernodePublicId(supernode_public_id),
+              SupernodePublicAddress(supernode_public_address) {}
+          bool operator ==(const GraftStakeTxExtra &rhs) const {
+            return SupernodePublicId == rhs.SupernodePublicId &&
+              !memcmp(&SupernodePublicAddress.m_view_public_key.data[0], &rhs.SupernodePublicAddress.m_view_public_key.data[0],
+                sizeof(SupernodePublicAddress.m_view_public_key.data));
+          }
+
+          BEGIN_SERIALIZE_OBJECT()
+              FIELD(SupernodePublicId)
+              FIELD(SupernodePublicAddress)
+          END_SERIALIZE()
```

Comment:
This doesn't seem like enough information: it also crucially needs a supernode signature so that someone cannot hijack a supernode.

Reply:
Could you please describe how a supernode can be hijacked?

Reply:
Hijack was perhaps too strong a node; what I had in mind: you set up a supernode, but before you submit the stake tx someone else submits a stake to your supernode, thus redirecting the rewards into their wallet while leaving you with the operating expenses (assuming it goes unnoticed).  Including a signature would ensure that only the actual operator can stake.

Reply:
As for me hijack potentially useless, since supernode knows its public wallet address (it gets it from its settings, not from the stake transaction). Therefore, this supernode will reject RTA transaction since it won't find the fee for its public wallet address. However, this hijack will create the wrong state of the network, when a hijacked supernode thinks that it doesn't have an active stake, but nodes and other supernodes in the network think that this supernode has an active stake.

As result, we think that adding a signature for public identification key and public wallet address can be a good idea. Current flow is next: supernode owner asks public identification key from supernode (which already knows its public wallet address), supernode generate the signature for public identification key and public wallet address using the private identification key and returns them to supernode owner, supernode owner gets public identification key, public wallet address and signature, and creates transaction with them (signature will be added to transaction). We'll update related RFCs, and, of course, this PR.

Reply:
Supernode signature has been added to stake transaction.